.github/workflows/ci.yml

```diff
@@ -46,26 +56,37 @@ jobs:
     name: plugin-test (pnpm v${{ matrix.version }}, Node.js v${{ matrix.node }}, ${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     steps:
+      - uses: actions/github-script@v7
+        id: calculate_architecture
+        with:
+          result-encoding: string
+          script: |
+            if ('${{ matrix.os }}' === 'macos-latest' && ['10', '12', '14'].includes('${{ matrix.node }}')) {
+              return "x64"
+            } else {
+              return ''
+            }
```

Comment:
Some context:
- macos-latest now runs on ARM chips https://github.com/actions/runner-images/issues/9741#issuecomment-2075259811.
- no ARM version for node 10, 12, 14 is [available](https://github.com/actions/node-versions/releases/tag/10.24.1-725060485), also see https://github.com/actions/setup-node/issues/1017#issuecomment-2077362713.

So we fallback to x64 on those versions. 

Step is copied from webpack: https://github.com/webpack/webpack/pull/18359